lms/djangoapps/instructor/views/api.py

```diff
@@ -2437,6 +2442,226 @@ def problem_grade_report(request, course_id):
+def get_assignment_names(request, course_id):
+    """
+    Returns a datatable of the assignments available for this course
+    """
+    course_id = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+    course = get_course_by_id(course_id)
+    assignment_names = get_course_assignment_labels(course)
+    return JsonResponse({
+        'data': assignment_names
+    })
+
+
+@require_POST
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('staff')
+def list_remote_assignments(request, course_id):
+    """
+    Returns a datatable of the assignments available in the remote gradebook
+    """
+    course_id = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+    course = get_course_by_id(course_id)
+    error_msg, datatable = _do_remote_gradebook_datatable(request.user, course, 'get-assignments')
+    datatable['title'] = _('Remote Gradebook Assignments')
+    return JsonResponse({
+        'errors': error_msg,
+        'datatable': datatable,
+    })
+
+
+def create_datatable_csv(csv_file, datatable):
+    """Creates a CSV file from the contents of a datatable."""
+    writer = csv.writer(csv_file, dialect='excel', quotechar='"', quoting=csv.QUOTE_ALL)
+    encoded_row = [unicode(s).encode('utf-8') for s in datatable['header']]
+    writer.writerow(encoded_row)
+    for datarow in datatable['data']:
+        # 's' here may be an integer, float (eg score) or string (eg student name)
+        encoded_row = [
+            # If s is already a UTF-8 string, trying to make a unicode
+            # object out of it will fail unless we pass in an encoding to
+            # the constructor. But we can't do that across the board,
+            # because s is often a numeric type. So just do this.
+            s if isinstance(s, str) else unicode(s).encode('utf-8')
+            for s in datarow
+        ]
+        writer.writerow(encoded_row)
+    return csv_file
+
+
+@require_POST
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('staff')
+def display_assignment_grades(request, course_id):
+    """
+    Returns a datatable of students' grades for an assignment in a course that matches a given course id
+    """
+    course_id = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+    course = get_course_by_id(course_id)
+    error_msg, datatable = _get_assignment_grade_datatable(course, request.POST.get('assignment_name', ''))
+    return JsonResponse({
+        'errors': error_msg,
+        'datatable': datatable
+    })
+
+
+@require_POST
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('staff')
+def export_assignment_grades_to_rg(request, course_id):
+    """
+    Exports students' grades for an assignment to the remote gradebook, then returns a
+    datatable of those grades
+    """
+    course_id = SlashSeparatedCourseKey.from_deprecated_string(course_id)
```

Comment:
I think that sounds reasonable. Seems like it's the most future-proof option while still preserving the ability to work with 'old mongo' style course ids.